stan/math/rev/core/vari.hpp

```diff
@@ -300,12 +314,115 @@ class vari_value<T, require_eigen_dense_base_t<T>> : public vari_base {
    */
   inline void init_dependent() { adj_.setOnes(); }
 
+  /**
+   * Implimentation for assignable objects that sets the adjoint values to zero.
+   */
+  template <typename Checker = T,
+            require_t<bool_constant<!std::is_const<Checker>::value>>* = nullptr>
+  inline void set_zero_adjoint_impl() {
+    adj_.setZero();
+  }
+  /**
+   * Implimentation for views which performs a no-op
+   */
+  template <typename Checker = T, require_t<std::is_const<Checker>>* = nullptr>
+  inline void set_zero_adjoint_impl() {}
+
   /**
    * Set the adjoint value of this variable to 0.  This is used to
    * reset adjoints before propagating derivatives again (for
    * example in a Jacobian calculation).
    */
-  inline void set_zero_adjoint() final { adj_.setZero(); }
+  inline void set_zero_adjoint() final { set_zero_adjoint_impl(); }
+
+  /**
+   * A block view of the underlying Eigen matrices.
+   * @param i Starting row of block.
+   * @param j Starting columns of block.
+   * @param p Number of rows to return.
+   * @param q Number of columns to return.
+   */
+  inline const auto block(Eigen::Index i, Eigen::Index j, Eigen::Index p,
+                          Eigen::Index q) const {
```

Comment:
```suggestion
  inline const auto block(Eigen::Index row, Eigen::Index col, Eigen::Index rows,
                          Eigen::Index cols) const {
```

Can you please not use single-letter variable names? I don't mind `i` for loop variables, or `x` and `y` for coordinates or `n` for number, but those `p` and `q` are really not meaningful.

Reply:
Yeah I don't like them either but was just following the Eigen names on these. Agree on changing the names tho'